server/segment_rpc.go

```diff
@@ -59,24 +59,30 @@ func (h *lphttp) ServeSegment(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// oInfo will be non-nil if we need to send an updated net.OrchestratorInfo
-	// to the broadcaster
+	// oInfo will be non-nil if we need to send an updated net.OrchestratorInfo to the broadcaster
 	var oInfo *net.OrchestratorInfo
 
-	if err := orch.ProcessPayment(payment, segData.ManifestID); err != nil {
-		glog.Errorf("Error processing payment: %v", err)
+	if paymentError := orch.ProcessPayment(payment, segData.ManifestID); paymentError != nil {
 
+		acceptableErr, ok := paymentError.(core.AcceptableError)
+		if !ok || !acceptableErr.Acceptable() {
```

Comment:
Noting that this will result in a HTTP error when `payment.Sender = nil || len(payment.Sender) == 0` and when `payment.TicketParams = nil` which would be the case for empty payment headers. While we don't send empty payments right now (since B usually doesn't credit its balance past the minimum), it might be useful to allow empty payments as long as the balance is sufficient and the price is acceptable (small change to `if ok && !acceptableErr.Acceptable()`)

Reply:
Fixed in aceab05